File: regular_assignments/day_006/sum_of_max_min/solution.rb
class Solution
  def call
    solve([1,2,3,4,5]) # 6
  end

  def solve(arr)
    inf = 10000000000
    min = inf
    max = -inf
    i = 0
    while i < arr.length
      min = find_min(arr[i], min)
      max = find_max(arr[i], max)
      i += 1
    end
    max + min
  end

  def find_min(a, b)
    a > b ? b : a
  end

  def find_max(a, b)
    a > b ? a : b
  end
end

File: regular_assignments/day_006/sum_of_max_min/solution.go
package sumOfMinMax

func Call() {
	solve([]int{1, 2, 3, 4, 5})
}

func solve(arr []int) int {
	inf := 10000000000
	min := inf
	max := -inf
	for i := 0; i < len(arr); i++ {
		min = findMin(arr[i], min)
		max = findMax(arr[i], max)
	}
	return max + min
}

func findMin(a, b int) int {
	if a > b {
		return b
	} else {
		return a
	}
}

func findMax(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}
